rules/retrieve.smk

```diff
@@ -106,6 +94,234 @@ if (
+        log:
+            logs("retrieve_corine_primary.log"),
+        resources:
+            mem_mb=1000,
+        retries: 2
+        conda:
+            "../envs/environment.yaml"
+        script:
+            "../scripts/retrieve_corine_dataset_primary.py"
+
+
+if (H2_SALT_CAVERNS_DATASET := dataset_version("h2_salt_caverns"))["source"] in [
+    "archive"
+]:
+
+    rule retrieve_h2_salt_caverns:
+        input:
+            storage(
+                H2_SALT_CAVERNS_DATASET["url"],
+            ),
+        output:
+            f"{H2_SALT_CAVERNS_DATASET["folder"]}/h2_salt_caverns_GWh_per_sqkm.geojson",
+        retries: 2
+        run:
+            move(input[0], output[0])
+
+
+if (GDP_PER_CAPITA_DATASET := dataset_version("gdp_per_capita"))["source"] in [
+    "archive"
+]:
+
+    rule retrieve_gdp_per_capita:
+        input:
+            storage(
+                GDP_PER_CAPITA_DATASET["url"],
+            ),
+        output:
+            f"{GDP_PER_CAPITA_DATASET["folder"]}/GDP_per_capita_PPP_1990_2015_v2.nc",
+        retries: 2
+        run:
+            move(input[0], output[0])
+
+
+if (POPULATION_COUNT_DATASET := dataset_version("population_count"))["source"] in [
+    "archive",
+    "primary",
+]:
+
+    rule retrieve_population_count:
+        input:
+            storage(
+                POPULATION_COUNT_DATASET["url"],
+            ),
+        output:
+            f"{POPULATION_COUNT_DATASET["folder"]}/ppp_2019_1km_Aggregated.tif",
+        retries: 2
+        run:
+            move(input[0], output[0])
+
+            if POPULATION_COUNT_DATASET["source"] == "primary":
+                import xarray as xr
+                import rioxarray as rio
+
+                file_path = output[0]
+                ds = xr.open_dataarray(file_path)
+                ds_reqd = ds.sel(x=slice(15.55, 40.41), y=slice(52.49, 41.72))
+                ds_reqd.rio.to_raster(file_path)
+
+
+
+if (GHG_EMISSIONS_DATASET := dataset_version("ghg_emissions"))["source"] in [
+    "archive",
+    "primary",
+]:
+
+    rule retrieve_ghg_emissions:
+        params:
+            url=GHG_EMISSIONS_DATASET["url"],
+        input:
+            storage(
+                GHG_EMISSIONS_DATASET["url"],
+            ),
+        output:
+            f"{GHG_EMISSIONS_DATASET["folder"]}/UNFCCC_v23.csv",
+            zip=(
+                f"{GHG_EMISSIONS_DATASET["folder"]}/UNFCCC_v23.csv.zip"
+                if GHG_EMISSIONS_DATASET["source"] == "primary"
+                else []
+            ),
+            directory=(
+                directory(f"{GHG_EMISSIONS_DATASET["folder"]}")
+                if GHG_EMISSIONS_DATASET["source"] == "primary"
+                else []
+            ),
+        retries: 2
+        run:
+            move(input[0], output[0])
+            if GHG_EMISSIONS_DATASET["source"] == "primary":
+                handle_data_requests(params, output)
```

Comment:
Can we give this a similar treatment as the aquifier dataset and simplify? (to remove also the `handle_data_requests(...)` function and `params`?